server/configuration.go

```diff
@@ -209,8 +210,19 @@ func (p *Plugin) OnConfigurationChange() error {
 		return errors.Wrap(ensureBotError, "failed to ensure demo bot.")
 	}
 
+	config := p.API.GetConfig()
+	data, err := ioutil.ReadFile(*config.PluginSettings.Directory + "/com.mattermost.demo-plugin/assets/github.svg")
+	if err != nil {
+		return errors.Wrap(err, "failed to read bot image")
+	}
+
 	p.botId = botId
 
+	AppError := p.API.SetBotIconImage(botId, data)
```

Comment:
This will be called every time there is a config, lets check `p.API.GetBotIconImage` to see if the bot already has an icon set and then set if if it doesn't. It's kinda awkward doing that but will be addressed with https://mattermost.atlassian.net/browse/MM-17297